server/eventlog.go

```diff
@@ -53,6 +54,17 @@ func ParseMachineState(rawEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState,
 	}, createGroupedError("failed to fully parse MachineState:", errors)
 }
 
+func parseCanonicalEventLog(rawCanonicalEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState, error) {
+	cosState, err := getVerifiedCosState(rawCanonicalEventLog, pcrs)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.MachineState{
+		Cos: cosState,
+	}, err
+}
```

Comment:
Nit: Why not just have this be one method instead of two methods (where one doesn't do anything.

Specifically, why do we not just have the implementation of `getVerifiedCosState` live in `parseCanonicalEventLog`

Reply:
I am thinking maybe later if there are more canonical event log states other than the COS state, we can add them here.

Reply:
Ok that makes sense. If we are doing that split, then we might want to do something like:

In `parseCanonicalEventLog`:
  - Call DecodeToCEL
  - Do the PCR replay

Then have `getVerifiedCosState` only take a `cel.CEL`, and it would:
  - Loop though all the records
  - Create (and return) the container/cos state

IMO, this sepearates the COS specific stuff from the generic CEL stuff

Reply:
make sense, Done